src/main/java/sg/ncl/DataController.java

```diff
@@ -429,17 +430,146 @@ public String getPublicDatasets(Model model) {
+            message.append("<ul class=\"fa-ul\">");
+            for (ObjectError objectError : bindingResult.getAllErrors()) {
+                FieldError fieldError = (FieldError) objectError;
+                message.append("<li><i class=\"fa fa-exclamation-circle\"></i> ");
+                switch (fieldError.getField()) {
+                    case "fullName":
+                        message.append("You have to fill in your full name");
+                        break;
+                    case "email":
+                        message.append("You have to fill in your email address");
+                        break;
+                    case "jobTitle":
+                        message.append("You have to fill in your job title");
+                        break;
+                    case "institution":
+                        message.append("You have to fill in your institution");
+                        break;
+                    case "country":
+                        message.append("You have to fill in your country");
+                        break;
+                    default:
+                        message.append(fieldError.getField());
+                        message.append(" ");
+                        message.append(fieldError.getDefaultMessage());
+                }
+                message.append("</li>");
+            }
+            message.append("</ul>");
+            model.addAttribute(MESSAGE_ATTRIBUTE, message);
+            HttpEntity<String> dataRequest = createHttpEntityHeaderOnlyNoAuthHeader();
+            ResponseEntity dataResponse = restTemplate.exchange(properties.getPublicDataset(id), HttpMethod.GET, dataRequest, String.class);
+            String dataResponseBody = dataResponse.getBody().toString();
+            JSONObject dataInfoObject = new JSONObject(dataResponseBody);
+            Dataset dataset = extractDataInfo(dataInfoObject.toString());
+            model.addAttribute(DATASET, dataset);
+            return "data_public_id";
+        }
+
+        JSONObject puserObject = new JSONObject();
+        puserObject.put("fullName", puser.getFullName());
+        puserObject.put("email", puser.getEmail());
+        puserObject.put("jobTitle", puser.getJobTitle());
+        puserObject.put("institution", puser.getInstitution());
+        puserObject.put("country", puser.getCountry());
+
+        HttpEntity<String> request = createHttpEntityWithBodyNoAuthHeader(puserObject.toString());
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.getPublicDataUsers(), HttpMethod.POST, request, String.class);
+        String responseBody = response.getBody().toString();
+        log.info("Public user saved: {}", responseBody);
+        JSONObject object = new JSONObject(responseBody);
+        session.setAttribute(PUBLIC_USER_ID, object.getLong("id"));
+
+        return REDIRECT_DATA + "/public/" + id + "/resources";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "/resources" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)